crates/astria-sequencer/src/service/info.rs

```diff
@@ -36,28 +32,119 @@ use tendermint::{
+    pub(crate) const UNKNOWN_PATH: Self = Self(1);
+
+    pub(crate) fn info(self) -> Option<&'static str> {
+        match self.0 {
+            0 => Some("Ok"),
+            1 => Some("provided path is unknown"),
+            2 => Some("one or more path parameters were invalid"),
+            3 => Some("an internal server error occured"),
+            _ => None,
+        }
+    }
+}
+impl Display for AbciCode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.info().unwrap_or("<unknown abci code>"))
+    }
+}
+
+impl From<AbciCode> for Code {
+    fn from(value: AbciCode) -> Self {
+        value.0.into()
+    }
+}
 
 #[derive(Clone)]
 pub(crate) struct Info {
     storage: Storage,
+    query_router: matchit::Router<BoxedAbciQueryHandler>,
 }
 
 impl Info {
-    pub(crate) fn new(storage: Storage) -> Self {
-        Self {
+    pub(crate) fn new(storage: Storage) -> anyhow::Result<Self> {
+        let mut query_router = matchit::Router::new();
+        query_router
+            .insert(
+                "accounts/balance/:account",
+                BoxedAbciQueryHandler::from_handler(crate::accounts::query::balance_request),
+            )
+            .context("invalid path: `accounts/balance/:account`")?;
+        query_router
+            .insert(
+                "accounts/nonce/:account",
+                BoxedAbciQueryHandler::from_handler(crate::accounts::query::nonce_request),
+            )
+            .context("invalid path: `accounts/nonce/:account`")?;
+        Ok(Self {
             storage,
+            query_router,
+        })
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_info_request(self, request: InfoRequest) -> Result<InfoResponse, BoxError> {
+        match request {
+            InfoRequest::Info(_) => {
+                let response = InfoResponse::Info(response::Info {
+                    version: "0.1.0".to_string(),
+                    app_version: 1,
+                    last_block_height: Height::default(),
+                    last_block_app_hash: AppHash::default(),
+                    data: "astria_sequencer".to_string(),
+                });
+                Ok(response)
+            }
+            InfoRequest::Echo(echo) => Ok(InfoResponse::Echo(Echo {
+                message: echo.message,
+            })),
+            InfoRequest::Query(req) => Ok(InfoResponse::Query(self.handle_abci_query(req).await)),
+            // this was removed after v0.34
+            InfoRequest::SetOption(_) => Ok(InfoResponse::SetOption(SetOption {
+                code: Code::default(),
+                log: "SetOption is not supported".to_string(),
+                info: "SetOption is not supported".to_string(),
+            })),
         }
     }
+
+    /// Handles `abci_query` RPCs.
+    async fn handle_abci_query(self, request: request::Query) -> response::Query {
+        let (handler, params) = match self.query_router.at(&request.path) {
```

Comment:
where do the `params` come from here? are they just parsed from the path basically?